### PR TITLE
[35961] Decrypt incoming messages

### PR DESCRIPTION
**Pull request by wesley4869**
Decrypt the eBL in JWE format

### CODE REVIEW

File: pkg/envelope/jwe.go [hunk 1]
```diff
@@ -33,15 +35,13 @@ func Encrypt(payload []byte, enc ContentEncryptionAlgorithm, keySettings []KeyEn
...
-		for _, r := range in.Recipients {
-			options = append(options, jwe.WithKey(jwa.KeyEncryptionAlgorithm(r.Header.Alg), key))
-		}
-		if len(in.Recipients) == 0 {
-			options = append(options, jwe.WithKey(jwa.KeyEncryptionAlgorithm(in.Header.Alg), key))
+		switch key.(type) {
+		case *rsa.PrivateKey:
+			options = append(options, jwe.WithKey(jwa.RSA_OAEP_256, key))
+		case *ecdsa.PrivateKey:
+			options = append(options, jwe.WithKey(jwa.ECDH_ES_A256KW, key))
```
Comment (EnnioLai) [inline pkg/envelope/jwe.go:44]: Based on the key type to choose EncryptionAlgorithm is wrong. 
We should put the keys into options when the key is compatible with r.Header.Alg. 
For example, if r.Header.Alg is ECDH_ES, ECDH_ES_A128KW or other ECDH_ES series , the compatible key is ECDSA key.
RSA1_5, RSA_OAEP or RSA_OAEP_256 are compatible to RSA key.

File: pkg/bu_server/broker/broker_test.go [hunk 2]
```diff
@@ -206,3 +289,27 @@ func loadTradeDocument(datafile string) storage.TradeDocument {
...
+	if ecdsaErr != nil && rsaErr != nil {
+		panic("failed to parse private key")
+	}
+	if ecdsaKey != nil {
+		return ecdsaKey
+	}
+	if rsaKey != nil {
+		return rsaKey
+	}
+	panic("no private key parsed")
```
Comment (EnnioLai) [inline pkg/bu_server/broker/broker_test.go:314]: pkg/pkix/cert.go ParsePrivateKey can help you to read private key from PEM file.

File: pkg/envelope/jwe.go (hunk 1 above)
Comment (wesley4869) [reply pkg/envelope/jwe.go]: Fixed.
Create options against Recipients and Header algorithms.

File: pkg/bu_server/broker/broker_test.go (hunk 2 above)
Comment (wesley4869) [reply pkg/bu_server/broker/broker_test.go]: Updated.

**Pull request merged**